fastapi_sqla/async_sqla.py

```diff
@@ -154,3 +139,38 @@ async def get_users(session: fastapi_sqla.AsyncSession = Depends()):
             await session.rollback()
 
     return response
+
+
+class AsyncSessionDependency:
+    def __init__(self, key: str = _DEFAULT_SESSION_KEY) -> None:
+        self.key = key
+
+    def __call__(self, request: Request) -> SqlaAsyncSession:
+        """Yield the sqlalchemy async session for that request.
+
+        It is meant to be used as a FastAPI dependency::
+
+            from fastapi import APIRouter, Depends
+            from fastapi_sqla import SqlaAsyncSession, AsyncSessionDependency
+
+            router = APIRouter()
+
+            @router.get("/users")
+            async def get_users(
+                session: SqlaAsyncSession = Depends(AsyncSessionDependency())
+            ):
+                pass
+        """
+        try:
+            return request.scope[f"{_ASYNC_REQUEST_SESSION_KEY}_{self.key}"]
+        except KeyError:
+            logger.exception(
+                f"No async session with key '{self.key}' found in request, "
+                "please ensure you've setup fastapi_sqla.",
+                session_key=self.key,
+            )
+            raise
+
+
+default_async_session_dep = AsyncSessionDependency()
```

Comment:
Addresses [this comment](https://github.com/dialoguemd/fastapi-sqla/pull/105#issuecomment-1795317685) for easier overrides